project/partners/templates/partners/index.html

```diff
@@ -0,0 +1,55 @@
+{% extends 'base.html' %}  
+{% block title %}Partners{% endblock %}
+{% block description %}This is the partners page (with leaflet){% endblock %}
+{% block content %}
+<section class="content">
+    <article class="sponsors">
+        {% for id, category in partners.items %}
+            <h2 class="page-heading">{{ category.title }}</h2>
+            <div class="category container-fluid p-0" id="{{ id }}">
+                {% for partner in category.items %}
+                    <div class="child">
+                        {% load static %}
+                        <div class="card-container">
+                                <div class="container-image">
+                                    <a href="{{ partner.link }}">
+                                        <img src="{% static "images/tedxntua-black-600.png" %}" alt="random" class="logo">                           
+                                    </a>
+                                    {% if partner.leaflet %}
+                                        <a href="{{ partner.link }}" class="title-right">
+                                            <button class="btn btn-partners btn-right"><h4>WEBSITE</h4></button> 
+                                        </a>
+                                    {% else %}
+                                        <a href="{{ partner.link }}" class="title-right">
+                                            <button class="btn btn-partners btn-right"></button> 
+                                        </a>
+                                    {% endif %}
+                                    
+                                    {% if partner.leaflet %}
+                                        <a href="#myModal" class="title-bottom" data-toggle="modal">
+                                            <button class="btn btn-partners btn-bottom"><h4>LEAFLET</h4></button>
+                                        </a>
+                                        <!-- Modal Image -->
+                                        <div class="modal fade bd-example-modal-lg show" id="myModal" role="dialog" tabindex="-1">
```

Comment:
Εδώ μάλλον θα χρειαστεί Javascript.
Βγάζει πιο πολύ νόημα να έχουμε ένα modal στη σελίδα (αντί για ένα modal για κάθε χορηγό), του οποίου το περιεχόμενο θα αλλάζει όταν κάνει κλικ ο χρήστης στο κουμπί, για τρεις λόγους:
1. Λιγότερο παραγόμενο html
2. Βοηθάει στη φόρτωση των εικόνων, γιατί όπως είναι τώρα θα φορτώνονται όλες οι εικόνες μόλις κάποιος μπαίνει στη σελίδα (καθώς ο browser θα βλέπει όλα τα `img` tags χωρίς να ξέρει ότι δεν φαίνονται στον χρήστη).
3. Δεν πρέπει να υπάρχουν πάνω από ένα elements σε μία σελίδα HTML που να έχουν το ίδιο id (`myModal`). Θα μπορούσαμε εναλλακτικά να κάνουμε κάτι τύπου (`id="myModal{{partner.id}}"`) για να βγαίνουν unique ids.

Χρήσιμα:
https://getbootstrap.com/docs/4.4/components/modal/#varying-modal-content

Ο κώδικας Javascript θα μπορούσε να είναι κάπως έτσι (θα το δοκιμάσω και θα σου πω):
```js
$('#myModal').on('show.bs.modal', function (event) {
  var btn = event.relatedTarget
  var leafletPath = btn.dataset.leafletPath
  // Update img of modal
  document.getElementById('#leaflet-image').setAttribute('src', leafletPath)
})
```

Reply:
na to koitakso tora i tha to koitaksoume to apogeuma?